### PR TITLE
doc(install): add Longhorn CLI in quick installation section

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#9075

#### What this PR does / why we need it:

Users can understand how to use the Longhorn CLI tool to check/install/set up the environment before installing the Longhorn.

### CODE REVIEW

File: content/docs/1.7.0/deploy/install/_index.md [hunk 1]
```diff
@@ -54,6 +54,51 @@ You must perform additional setups before using Longhorn with certain operating
 - Talos Linux clusters: See [Talos Linux Support](../../advanced-resources/os-distro-specific/talos-linux-support).
 - Container-Optimized OS: See [Container-Optimized OS Support](../../advanced-resources/os-distro-specific/container-optimized-os-support).
 
+### Using the Longhorn Command Line Tool
+
+The `longhornctl` tool is a CLI interface to Longhorn operations to help you gather enough information about the factors. For more information, please see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
+
+Download the `longhornctl` tool binary file and use the `check` sub-command to identifies potential issues before usage before installing Longhorn：
```
Comment (derekbit) [inline content/docs/1.7.0/deploy/install/_index.md:61]:
```suggestion
To check the prerequisites and configurations, please download the command line tool `longhornctl` and utilize the `check` sub-command:
```

Comment (mantissahz) [reply content/docs/1.7.0/deploy/install/_index.md]: Modified. Thanks.

File: content/docs/1.7.0/deploy/install/_index.md [hunk 2]
```diff
@@ -54,6 +54,51 @@ You must perform additional setups before using Longhorn with certain operating
 - Talos Linux clusters: See [Talos Linux Support](../../advanced-resources/os-distro-specific/talos-linux-support).
 - Container-Optimized OS: See [Container-Optimized OS Support](../../advanced-resources/os-distro-specific/container-optimized-os-support).
 
+### Using the Longhorn Command Line Tool
+
+The `longhornctl` tool is a CLI interface to Longhorn operations to help you gather enough information about the factors. For more information, please see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
```
Comment (derekbit) [inline content/docs/1.7.0/deploy/install/_index.md:59]:
```suggestion
The `longhornctl` tool is a CLI interface to Longhorn operations. For more information, please see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
```

**Review by derekbit — COMMENTED** [review]
BTW, also need to update [v2 volume Quick Start ](https://longhorn.io/docs/1.7.0/v2-data-engine/quick-start/). Add `--enable-spdk` to the two sub-commands.

File: content/docs/1.7.0/deploy/install/_index.md (hunk 2 above)
Comment (mantissahz) [reply content/docs/1.7.0/deploy/install/_index.md]: Modified. Thanks.

>BTW, also need to update [v2 volume Quick Start ](https://longhorn.io/docs/1.7.0/v2-data-engine/quick-start/). Add --enable-spdk to the two sub-commands.

Updated. Thanks

File: content/docs/1.7.0/deploy/install/_index.md [hunk 3]
```diff
@@ -54,6 +54,51 @@ You must perform additional setups before using Longhorn with certain operating
 - Talos Linux clusters: See [Talos Linux Support](../../advanced-resources/os-distro-specific/talos-linux-support).
 - Container-Optimized OS: See [Container-Optimized OS Support](../../advanced-resources/os-distro-specific/container-optimized-os-support).
 
+### Using the Longhorn Command Line Tool
+
+The `longhornctl` tool is a CLI interface to Longhorn operations. For more information, please see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
+
+To check the prerequisites and configurations, please download the command line tool `longhornctl` and utilize the `check` sub-command:
```
Comment (jillian-maroket) [inline content/docs/1.7.0/deploy/install/_index.md:61]:
```suggestion
The `longhornctl` tool is a CLI interface to Longhorn operations. For more information, see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).

To check the prerequisites and configurations, download the tool and run the `check` sub-command:
```

File: content/docs/1.7.0/deploy/install/_index.md [hunk 4]
```diff
@@ -54,6 +54,51 @@ You must perform additional setups before using Longhorn with certain operating
...
+# For AMD64 platform
+curl -sSfL -o longhornctl https://github.com/longhorn/cli/releases/download/v{{< current-version >}}/longhornctl-linux-amd64
+# For ARM platform
+curl -sSfL -o longhornctl https://github.com/longhorn/cli/releases/download/v{{< current-version >}}/longhornctl-linux-arm64
+
+chmod +x longhornctl
+./longhornctl check preflight
+```
+
+Example result:
```
Comment (jillian-maroket) [inline content/docs/1.7.0/deploy/install/_index.md:73]:
```suggestion
Example of result:
```

File: content/docs/1.7.0/deploy/install/_index.md [hunk 5]
```diff
@@ -54,6 +54,51 @@ You must perform additional setups before using Longhorn with certain operating
...
+  - multipathd.service is running. Please refer to https://longhorn.io/kb/troubleshooting-volume-with-multipath/ for more information.
+worker2:
+  info:
+  - Service iscsid is running
+  - NFS4 is supported
+  - Package nfs-common is not installed
+  - Package open-iscsi is installed
+```
+
+You can use the `install` sub-command to perform preflight dependencies installation and setup before installing Longhorn.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/deploy/install/_index.md:96]:
```suggestion
Use the `install` sub-command to install and set up the preflight dependencies before installing Longhorn.
```

File: content/docs/1.7.0/v2-data-engine/quick-start.md [hunk 6]
```diff
@@ -105,6 +107,79 @@ After finishing the above steps, restart kubelet on each node.
 
 ### Check Environment
 
+#### With The Longhorn Command Line Tool
+
+The `longhornctl` tool is a CLI interface to Longhorn operations. For more information, please see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
+
+To check the prerequisites and configurations, please download the command line tool `longhornctl` and utilize the `check` sub-command:
```
Comment (jillian-maroket) [inline content/docs/1.7.0/v2-data-engine/quick-start.md:114]:
```suggestion
The `longhornctl` tool is a CLI interface to Longhorn operations. For more information, see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).

To check the prerequisites and configurations, download the tool and run the `check` sub-command:
```

File: content/docs/1.7.0/v2-data-engine/quick-start.md [hunk 7]
```diff
@@ -105,6 +107,79 @@ After finishing the above steps, restart kubelet on each node.
...
+# For AMD64 platform
+curl -sSfL -o longhornctl https://github.com/longhorn/cli/releases/download/v{{< current-version >}}/longhornctl-linux-amd64
+# For ARM platform
+curl -sSfL -o longhornctl https://github.com/longhorn/cli/releases/download/v{{< current-version >}}/longhornctl-linux-arm64
+
+chmod +x longhornctl
+./longhornctl check preflight --enable-spdk
+```
+
+Example result:
```
Comment (jillian-maroket) [inline content/docs/1.7.0/v2-data-engine/quick-start.md:126]:
```suggestion
Example of result:
```

File: content/docs/1.7.0/v2-data-engine/quick-start.md [hunk 8]
```diff
@@ -105,6 +107,79 @@ After finishing the above steps, restart kubelet on each node.
...
+  - Service iscsid is running
+  - NFS4 is supported
+  - Package nfs-common is installed
+  - Package open-iscsi is installed
+  - CPU instruction set sse4_2 is supported
+  warn:
+  - multipathd.service is running. Please refer to https://longhorn.io/kb/troubleshooting-volume-with-multipath/ for more information.
+```
+
+Use the `install` sub-command to perform preflight dependencies installation and setup before installing Longhorn.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/v2-data-engine/quick-start.md:148]:
```suggestion
Use the `install` sub-command to install and set up the preflight dependencies before installing Longhorn.
```

File: content/docs/1.7.0/v2-data-engine/quick-start.md [hunk 9]
```diff
@@ -105,6 +107,79 @@ After finishing the above steps, restart kubelet on each node.
...
+INFO[2024-01-01T00:00:03Z] Initializing preflight installer
+INFO[2024-01-01T00:00:03Z] Cleaning up preflight installer
+INFO[2024-01-01T00:00:03Z] Running preflight installer
+INFO[2024-01-01T00:00:03Z] Installing dependencies with package manager
+INFO[2024-01-01T00:00:10Z] Installed dependencies with package manager
+INFO[2024-01-01T00:00:10Z] Cleaning up preflight installer
+INFO[2024-01-01T00:00:10Z] Completed preflight installer. Use 'longhornctl check preflight' to check the result.
+```
+
+After executing the `install` sub-command, you can use the `check` sub-command again to ensure all environment settings are correct:
```
Comment (jillian-maroket) [inline content/docs/1.7.0/v2-data-engine/quick-start.md:161]:
```suggestion
After installing and setting up the preflight dependencies, you can run the `check` sub-command again to verify that all environment settings are correct.
```

File: content/docs/1.7.0/v2-data-engine/quick-start.md [hunk 10]
```diff
@@ -105,6 +107,79 @@ After finishing the above steps, restart kubelet on each node.
 
 ### Check Environment
 
+#### With The Longhorn Command Line Tool
```
Comment (jillian-maroket) [inline content/docs/1.7.0/v2-data-engine/quick-start.md:110]:
```suggestion
#### Using the Longhorn Command Line Tool
```

File: content/docs/1.7.0/v2-data-engine/quick-start.md [hunk 11]
```diff
@@ -105,6 +107,79 @@ After finishing the above steps, restart kubelet on each node.
...
+  - NFS4 is supported
+  - Package nfs-common is installed
+  - Package open-iscsi is installed
+  - CPU instruction set sse4_2 is supported
+  - HugePages is enabled
+  - Module nvme_tcp is loaded
+  - Module uio_pci_generic is loaded
+```
+
+#### With The Script
```
Comment (jillian-maroket) [inline content/docs/1.7.0/v2-data-engine/quick-start.md:181]:
```suggestion
#### Using the Script
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

File: content/docs/1.7.0/deploy/install/_index.md (hunk 3 above)
Comment (mantissahz) [reply content/docs/1.7.0/deploy/install/_index.md]: Modified. Thanks.

File: content/docs/1.7.0/deploy/install/_index.md (hunk 4 above)
Comment (mantissahz) [reply content/docs/1.7.0/deploy/install/_index.md]: Modified. Thanks.

File: content/docs/1.7.0/deploy/install/_index.md (hunk 5 above)
Comment (mantissahz) [reply content/docs/1.7.0/deploy/install/_index.md]: Modified. Thanks.

File: content/docs/1.7.0/v2-data-engine/quick-start.md (hunk 6 above)
Comment (mantissahz) [reply content/docs/1.7.0/v2-data-engine/quick-start.md]: Modified. Thanks.

File: content/docs/1.7.0/v2-data-engine/quick-start.md (hunk 7 above)
Comment (mantissahz) [reply content/docs/1.7.0/v2-data-engine/quick-start.md]: Modified. Thanks.

File: content/docs/1.7.0/v2-data-engine/quick-start.md (hunk 8 above)
Comment (mantissahz) [reply content/docs/1.7.0/v2-data-engine/quick-start.md]: Modified. Thanks.

File: content/docs/1.7.0/v2-data-engine/quick-start.md (hunk 9 above)
Comment (mantissahz) [reply content/docs/1.7.0/v2-data-engine/quick-start.md]: Modified. Thanks.

File: content/docs/1.7.0/v2-data-engine/quick-start.md (hunk 10 above)
Comment (mantissahz) [reply content/docs/1.7.0/v2-data-engine/quick-start.md]: Modified. Thanks.

File: content/docs/1.7.0/v2-data-engine/quick-start.md (hunk 11 above)
Comment (mantissahz) [reply content/docs/1.7.0/v2-data-engine/quick-start.md]: Modified. Thanks.

File: content/docs/1.7.0/deploy/install/_index.md [hunk 12]
```diff
@@ -54,6 +54,51 @@ You must perform additional setups before using Longhorn with certain operating
 - Talos Linux clusters: See [Talos Linux Support](../../advanced-resources/os-distro-specific/talos-linux-support).
 - Container-Optimized OS: See [Container-Optimized OS Support](../../advanced-resources/os-distro-specific/container-optimized-os-support).
 
+### Using the Longhorn Command Line Tool
+
+The `longhornctl` tool is a CLI interface to Longhorn operations. For more information, see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
```
Comment (jhkrug) [inline content/docs/1.7.0/deploy/install/_index.md:59]: Removed a redundant 'interface'.
```suggestion
The `longhornctl` tool is a CLI for Longhorn operations. For more information, see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
```

File: content/docs/1.7.0/v2-data-engine/quick-start.md [hunk 13]
```diff
@@ -105,6 +107,79 @@ After finishing the above steps, restart kubelet on each node.
 
 ### Check Environment
 
+#### Using the Longhorn Command Line Tool
+
+The `longhornctl` tool is a CLI interface to Longhorn operations. For more information, see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
```
Comment (jhkrug) [inline content/docs/1.7.0/v2-data-engine/quick-start.md:112]: Likewise.
```suggestion
The `longhornctl` tool is a CLI for Longhorn operations. For more information, see [Command Line Tool (longhornctl)](../../advanced-resources/longhornctl/).
```

**Review by jhkrug — CHANGES_REQUESTED** [review]
A minor suggestion

File: content/docs/1.7.0/deploy/install/_index.md (hunk 12 above)
Comment (mantissahz) [reply content/docs/1.7.0/deploy/install/_index.md]: Modified. Thanks.

**Review by derekbit — APPROVED** [review]

**Pull request merged**